src/main/java/com/zaxxer/hikari/metrics/prometheus/PrometheusMetricsTrackerFactory.java

```diff
@@ -19,29 +19,91 @@
  * <pre>{@code
  * HikariConfig config = new HikariConfig();
  * config.setMetricsTrackerFactory(new PrometheusMetricsTrackerFactory());
  * }</pre>
+ * </p>
+ * <p>
+ * Use provided CollectorRegistry and default HikariCPCollector:
+ * <pre>{@code
+ * HikariConfig config = new HikariConfig();
+ * config.setMetricsTrackerFactory(new PrometheusMetricsTrackerFactory(registry));
+ * }</pre>
+ * </p>
+ * <p>
+ * Use provided CollectorRegistry and provided HikariCPCollector:
+ * <pre>{@code
+ * HikariConfig config = new HikariConfig();
+ * config.setMetricsTrackerFactory(new PrometheusMetricsTrackerFactory(registry, collector));
+ * }</pre>
+ * </p>
  */
 public class PrometheusMetricsTrackerFactory implements MetricsTrackerFactory {
 
-   private static HikariCPCollector collector;
+   private HikariCPCollector collector;
+   private CollectorRegistry registry;
+
+   private static volatile HikariCPCollector globalCollector;
+
+   /**
+    * If nothing is provided, use {@link CollectorRegistry.defaultRegistry} and {@link globalCollector} as default
+    */
+   public PrometheusMetricsTrackerFactory() {
+      this(CollectorRegistry.defaultRegistry);
+   }
+
+   /**
+    * If a CollectorRegistry is provided, use provided CollectorRegistry and {@link globalCollector} as default
+    */
+   public PrometheusMetricsTrackerFactory(CollectorRegistry registry) {
+      if (globalCollector == null) {
+         synchronized (PrometheusMetricsTrackerFactory.class) {
+            if (globalCollector == null) {
+               globalCollector = new HikariCPCollector();
+            }
+         }
+      }
+      this.registry = registry;
+      this.collector = globalCollector.register(registry);
+   }
+
+   /**
+    * If both a CollectorRegistry and a Collector are provided, use provided CollectorRegistry and Collector
+    */
+   public PrometheusMetricsTrackerFactory(CollectorRegistry registry, HikariCPCollector collector) {
+      this.registry = registry;
+      this.collector = collector.register(registry);
+   }
 
    @Override
    public IMetricsTracker create(String poolName, PoolStats poolStats) {
       getCollector().add(poolName, poolStats);
-      return new PrometheusMetricsTracker(poolName);
+      return new PrometheusMetricsTracker(poolName, registry);
+   }
+
+   public static void setGlobalCollector(HikariCPCollector globalCollector) {
+      PrometheusMetricsTrackerFactory.globalCollector = globalCollector;
    }
 
    /**
     * initialize and register collector if it isn't initialized yet
     */
    private HikariCPCollector getCollector() {
       if (collector == null) {
-         collector = new HikariCPCollector().register();
+         if (globalCollector == null) {
+            synchronized (PrometheusMetricsTrackerFactory.class) {
+               if (globalCollector == null) {
+                  globalCollector = new HikariCPCollector();
+               }
+            }
+         }
```

Comment:
This code is now identical to lines 65-71.  Refactor into a shared ``getGlobalCollector()`` method.